### PR TITLE
Issue #1181 Python 2 to 3 Progress (5 Scripts)

### PR DESCRIPTION
**Pull request by wallacejohn**
#### Description

I used the 2to3 tool to convert 5 Python 2 scripts to Python 3, and made a few additional minor changes to complete the upgrade for these scripts.

#### Related Issue

Issue #1181 

#### Motivation and Context

Python 2 is end-of-life; these changes remove dependency on Python 2.

#### How Has This Been Tested?

All changes were tested with Python 3.7.7 on Debian Testing. Each script was tested for syntax errors detected by the interpreter, followed by attempts to execute each script in its normal usage scenario to confirm that script behavior is unchanged. Finally, the basic and strong regression tests were run with no errors.

#### Types of changes

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: vtr_flow/scripts/download_ispd.py [hunk 1]
```diff
@@ -68,39 +68,41 @@ def main():
...
         md5_filename = "ispd_benchmarks_vtr_v" + args.version + '.md5'
         
-        tar_gz_url = urlparse.urljoin(ISPD_URL_MIRRORS[args.mirror], tar_gz_filename)
-        md5_url = urlparse.urljoin(ISPD_URL_MIRRORS[args.mirror], md5_filename)
+        tar_gz_url = urllib.parse.urljoin(ISPD_URL_MIRRORS[args.mirror], tar_gz_filename)
+        md5_url = urllib.parse.urljoin(ISPD_URL_MIRRORS[args.mirror], md5_filename)
 
-        external_md5 = load_md5_from_url(md5_url)
+        # Had to tack on a .decode() here as part of Python 2 to
+        # Python 3 conversion to convert from bytes to a string
```
Comment (kmurray) [inline vtr_flow/scripts/download_ispd.py]: I'd probably leave the fact that this is a Python 2 to 3 conversion issue to the commit messages, and just leave the second line (converting from bytes to string) in the comment.

File: vtr_flow/scripts/download_titan.py [hunk 2]
```diff
@@ -74,39 +74,41 @@ def main():
...
         md5_filename = "titan_release_" + args.titan_version + '.md5'
         
-        tar_gz_url = urlparse.urljoin(TITAN_URL_MIRRORS[args.mirror], tar_gz_filename)
-        md5_url = urlparse.urljoin(TITAN_URL_MIRRORS[args.mirror], md5_filename)
+        tar_gz_url = urllib.parse.urljoin(TITAN_URL_MIRRORS[args.mirror], tar_gz_filename)
+        md5_url = urllib.parse.urljoin(TITAN_URL_MIRRORS[args.mirror], md5_filename)
 
-        external_md5 = load_md5_from_url(md5_url)
+        # Had to tack on a .decode() here as part of Python 2 to
+        # Python 3 conversion to convert from bytes to a string
```
Comment (kmurray) [inline vtr_flow/scripts/download_titan.py]: Same as above

File: vtr_flow/scripts/upgrade_arch.py [hunk 3]
```diff
@@ -154,7 +154,12 @@ def main():
         if args.debug:
             root.write(sys.stdout, pretty_print=args.pretty)
         else:
-            with open(args.xml_file, "w") as f:
+            # Added the "b" flag to silence the str/bytes
+            # error. Output of script is identical except for some XML
+            # fields are in a different order. This seems to be the
+            # doing of the lxml library and won't be a simple fix if
+            # it causes problems.
```
Comment (kmurray) [inline vtr_flow/scripts/upgrade_arch.py]: Similarly, stating that the 'b' flag is needed for str/byte error is a good comment.

That the output is identical modulo lxml ordering is something that should go in the commit message. (Someone reading the comment years from now won't know what the reference point for 'identical' is, so best to leave that in the commit messages which give more temporal context).

**Comment by wallacejohn** [comment]
Thanks for the feedback. According to Python Developer PEP 394 (https://www.python.org/dev/peps/pep-0394/#recommendation), the recommendation is to have a more specific shebang when possible, so `python3` instead of `python` in our case. I updated those, plus cleaned up those comments. I'll follow a similar pattern for future scripts.

**Comment by kmurray** [comment]
Looks great! Thanks for putting this together.

**Comment by wallacejohn** [comment]
No problem! I'll get started on more scripts and submit them in groups of 5-10. Looks like there are ~50 Python scripts total throughout the project, although they may not all by Python 2.

**Pull request merged**